src/mesh_analysis/waku_message_log_analyzer.py

```diff
@@ -0,0 +1,132 @@
+
+        return Err(f'Unable to obtain pod name from {peer_id}')
+
+    def _get_affected_node_log(self, data_file: str):
+        result = self._get_affected_node_pod(data_file)
+        if result.is_err():
+            logger.warning(result.err_value)
+
+        victoria_config = {"url": "https://vmselect.riff.cc/select/logsql/query",
+                           "headers": {"Content-Type": "application/json"},
+                           "params": [{
+                               "query": f"kubernetes_pod_name:{result.ok_value} AND _time:{self._timestamp} | sort by (_time)"}]}
+
+        waku_tracer = WakuTracer()
+        waku_tracer.with_wildcard_pattern()
+        reader = VictoriaReader(victoria_config, waku_tracer)
+        pod_log = reader.read()
+
+        log_lines = [inner_list[0] for inner_list in pod_log[0]]
+        with open(self._dump_analysis_dir_path / f'{pod_log[0][0][1]}.log', 'w') as file:
+            for element in log_lines:
+                file.write(f"{element}\n")
+
+    def _dump_information(self, data_files: List[str]):
+        for data_file in data_files:
+            logger.info(f'Dumping information for {data_file}')
+            self._get_affected_node_log(data_file)
+
+    def _has_issues_in_local(self) -> bool:
+        waku_tracer = WakuTracer()
+        waku_tracer.with_received_pattern()
+        waku_tracer.with_sent_pattern()
+        reader = FileReader(self._local_folder_to_analyze_path, waku_tracer)
+        dfs = reader.read()
+
+        has_issues = waku_tracer.has_message_reliability_issues('msg_hash', 'receiver_peer_id', dfs[0], dfs[1],
+                                                                self._dump_analysis_dir_path)
+
+        return has_issues
+
+    def _has_issues_in_cluster(self) -> bool:
+        waku_tracer = WakuTracer()
+        waku_tracer.with_received_pattern()
+        waku_tracer.with_sent_pattern()
+        reader = VictoriaReader(self._victoria_config, waku_tracer)
+        dfs = reader.read()
+
+        has_issues = waku_tracer.has_message_reliability_issues('msg_hash', 'receiver_peer_id', dfs[0], dfs[1],
+                                                                self._dump_analysis_dir_path)
+
+        return has_issues
+
+    def analyze_message_logs(self):
+        if self._timestamp is not None:
+            has_issues = self._has_issues_in_cluster()
+            if has_issues:
+                match file_utils.get_files_from_folder_path(Path(self._dump_analysis_dir_path)):
+                    case Ok(data_files_names):
+                        self._dump_information(data_files_names)
+                    case Err(error):
+                        logger.error(error)
+        else:
+            has_issues = self._has_issues_in_local()
+
+    def analyze_message_timestamps(self, time_difference_threshold: int):
+        folder_path = self._local_folder_to_analyze_path or self._dump_analysis_dir_path
```

Comment:
Unlikely, but both of these could be `None`.

Reply:
No, this is ensured by `_validate_analysis_location` that is called in `__init__`.

Reply:
`_validate_analysis_location` contains this:
```py
if timestamp_to_analyze is None and local_folder_to_analyze is None:
    logger.error('No timestamp or local folder specified')
    exit(1)
```
This means if one of them is not `None` this will not exit. That is, if `timestamp_to_analyze` is not `None` the arm will be skipped. That means `local_folder_to_analyze` could be None and given `_local_folder_to_analyze_path` is not validated both of them could be `None`.

Reply:
I am not really following. If `timestamp_to_analyze` is not `None`, the folder will be handled automatically, because data will be dumped from Victoria Logs.
If you want to use local data that you don't need to pull, you can use `_local_folder_to_analyze_path`.
I am having a hard time finding where they can both be `None` with the use of `_validate_analysis_location ` and `folder_path = self._local_folder_to_analyze_path or self._dump_analysis_dir_path`.
If the local file doesn't exist, an error will be logged and returned.

Reply:
Talked in call, now I understand what you mean c: